scapy/libs/rfc3961.py

```diff
@@ -0,0 +1,729 @@
+# SPDX-License-Identifier: BSD-2-Clause
+# This file is part of Scapy
+# See https://scapy.net/ for more information
+# Copyright (c) 2013, Marc Horowitz
+
+"""
+Implementation of RFC 3961's cryptographic functions
+"""
+
+# The following is a modified version of
+# https://github.com/SecureAuthCorp/impacket/blob/3ec59074ec35c06bbd4312d1042f0e23f4a1b41f/impacket/krb5/crypto.py
+# itself heavily inspired from
+# https://github.com/mhorowitz/pykrb5/blob/master/krb5/crypto.py
+# Note that the following work is based only on THIS COMMIT from impacket,
+# which is therefore under mhorowitz's BSD 2-clause "simplified" license.
+
+import math
+import os
+import struct
+from scapy.compat import orb, chb, int_bytes, bytes_int, plain_str
+
+try:
+    from Cryptodome.Cipher import AES, DES3, ARC4, DES
```

Comment:
Cryptography doesn't support some of those (DES for instance), which is why we use cryptodome.. :/